src/utils/nameUtils.ts

```diff
@@ -143,7 +143,9 @@ export function normalizeName(
   const [firstPart, ...otherParts] = parts;
   const normalizedFirstPart = toCasing(firstPart, casingConvention);
   const normalizedParts = (otherParts || [])
-    .map(part => toCasing(part, CasingConvention.Pascal))
+    .map(part =>
+      part === "null" ? part : toCasing(part, CasingConvention.Pascal)
```

Comment:
Do you think there may be other scenarios where we'd like to keep the name? Would it be possible to have strings as `boolean`, `string`, `number` that would be impacted the same way `null` was?

If so, it may be worth using a collection of values that don't need to get a casing change and check that the current part is not present in that collection, instead of just checking for null

Reply:
This is a great feedback! I experimented with different ways to approach this. It turns out, `normalizeName` is a function that processes a single name and should not be called on a type expression. I updated `getTypeForSchema` to not call `normalizeTypeName` excessively and reverted `normalizeName` back.